### PR TITLE
[windows] implement PyPy checksum validation

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

implementing checksums for PyPy

#### Related issue:

https://github.com/github/c2c-actions-akvelon/issues/38

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/Installers/Install-PyPy.ps1 [hunk 1]
```diff
@@ -93,8 +100,23 @@ foreach($toolsetVersion in $toolsetVersions.versions)
...
     if ($latestMajorPyPyVersion)
     {
-        Write-Host "Found PyPy '$($latestMajorPyPyVersion.filename)' package"
-        $tempPyPyPackagePath = Start-DownloadWithRetry -Url $latestMajorPyPyVersion.download_url -Name $latestMajorPyPyVersion.filename
+        $filename = $latestMajorPyPyVersion.filename
+        Write-Host "Found PyPy '$filename' package"
+        $tempPyPyPackagePath = Start-DownloadWithRetry -Url $latestMajorPyPyVersion.download_url -Name $filename
+
+        #region Supply chain security
+        $local_file_hash = (Get-FileHash -Path $tempPyPyPackagePath -Algorithm SHA256).Hash
```
Comment (erik-bershel) [inline images/win/scripts/Installers/Install-PyPy.ps1:108]: I’m not sure myself, but it seems like a camelСase was expected to name the variables.

File: images/win/scripts/Installers/Install-PyPy.ps1 [hunk 2]
```diff
@@ -82,6 +83,12 @@ $toolsetVersions = Get-ToolsetContent | Select-Object -ExpandProperty toolcache
 # Get PyPy releases
 $pypyVersions = Invoke-RestMethod https://downloads.python.org/pypy/versions.json
 
+# required for html parsing
+Install-Module PowerHTML -Scope CurrentUser 
+Import-Module PowerHTML
+$checksums = (Invoke-RestMethod -Uri 'https://www.pypy.org/checksums.html' | ConvertFrom-HTML).SelectNodes('//*[@id="content"]/article/div/pre')
```
Comment (vpolikarpov-akvelon) [inline images/win/scripts/Installers/Install-PyPy.ps1:89]: This looks extremely fragile. Do we really need to parse HTML? Why not to treat page as plain text and search for proper "checksum-filename" pair instead?

Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-PyPy.ps1:89]: I do not agree with you. Page is html, treating it as plain text is more fragile from my point of view

File: images/win/scripts/Installers/Install-PyPy.ps1 (hunk 1 above)
Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-PyPy.ps1:108]: thank you for suggestion, much better indeed

File: images/win/scripts/Installers/Install-PyPy.ps1 (hunk 2 above)
Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-PyPy.ps1:89]: that's how parsing works. if that has changed, image generation will fail. that is expected.
I'm not writing uber parser that will survive such changes, I think it is fine

**Review by Alexey-Ayupov — APPROVED** [review]

**Pull request merged**